### ISSUE TITLE
uM element picker help [solved: PECAK]

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [ghacksuserjs/ghacks-user.js#480]
Hey guys (and gals)

This is driving me slightly nuts. It's not a problem per se, as in I can always look at the element code and craft a cosmetic filter, but it bugs me that the site will not allow the uM element picker (AFAICT).

Site: https://thechive.com/  and any sub-pages
Wot?: Try using the element picker - nothing happens
Wot2?: hidden elements do not collapse

![ubo](https://user-images.githubusercontent.com/16656956/43988612-05cf3914-9d29-11e8-8c02-b6331030d2da.png)
![um](https://user-images.githubusercontent.com/16656956/43988614-0ae2fdaa-9d29-11e8-820d-b3f48fb4b9c4.png)
![no-collapse](https://user-images.githubusercontent.com/16656956/43988615-0e5f98b2-9d29-11e8-9a30-f5e9ee25db9a.png)

Can someone figure it out. I'm not worried about the site, but I want to KNOW what the F is causing it, cuz I like to be in control and all that stuff, and I would hate other sites to start doing the same - currently this is the only site I found this on with all my default settings (which is only allow 1st party css + images)

@gorhill if you have some spare time, no hurry, it's probably just me, *but it may be you*, but it really probably is just me (and I'm an idiot and do not understand the interwebs very well)

**Comment by Thorin-Oakenpants** [comment]
PS: I know I don't need `r-login`, but it's one less rule to store

**Comment by Thorin-Oakenpants** [comment]
nevermind ... IDK why, but uBO was powered off (not a blue symbol) .. changed that and all is fine.

Still intrigued as to why effectively disabling uBO made uM not be able to use it's element picker

**Comment by Thorin-Oakenpants** [comment]
> Still intrigued as to why effectively disabling uBO made uM not be able to use it's element picker

@gorhill

**Comment by gorhill** [comment]
> disabling uBO made uM not be able to use it's element picker

There is no element picker in uMatrix, you probably think of uBO. The element picker is not available when uBO is disabled on a site, because the content script necessary for the picker to work are not present when uBO is disabled, and furthermore there is not much point to create a filter for a site for which uBO is disabled, the filter won't end up applying anyway.

**Comment by Thorin-Oakenpants** [comment]
aww man .. what a mess of a day. **STILL** mixing up uM and uBO for some reason. Now I feel like a right plonker (don't comment, all you trolls).

**Comment by earthlng** [comment]
> [solved: PECAK]

what does that mean?

**Comment by fmarier** [comment]
Probably a typo for [PEBCAK](https://www.urbandictionary.com/define.php?term=PEBCAK).

**Comment by Thorin-Oakenpants** [comment]
^^ exactly. Yesterday was just *one of those days* - couldn't get anything right!

My problem is twofold
1. Because I use the context menu (despite it having a uBO icon) instead of using the uBO panel to grab it, for some reason, in my head, I've associated it with uM.

2. Despite the fact that the toolbar uBO color was gray, despite looking at the uBO panel and the power icon was off, I did not even see that (partly because I was convinced I needed to look at uM!). And that stems from the UI in a sense.

I run a tight config, and sometimes when I need to get a site to work, I toggle uBO off in order to see all the domains, scripts, etc in uM. This can be problematic, because I have to remember to toggle it back on. I'm positive I have read a couple of issues on this, **but I would prefer when toggling the uBO power button, that it was temporary (session), and requires the user to hit the padlock or eraser to make permanent** - but I am sure there are pros/cons to both methods, and reasons why it is not.

I know I can check my whitelist, which I just did, to make sure I didn't leave any others - and I haven't.
